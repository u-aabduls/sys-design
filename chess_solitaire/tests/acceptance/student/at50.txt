-- ACCEPTANCE TEST #50: VIOLATION CASE: QUEEN BLOCKED - #7

-- Setup some chess pieces for the game
setup_chess (Q,1,4)
setup_chess (K,2,3)
setup_chess (P,3,2)
setup_chess (Q,4,1)

start_game

-- Blocked by King@(2,3)
move_and_capture(1,4,4,1)

-- Blocked by King@(2,3)
move_and_capture(1,4,3,2)

-- Blocked by King@(2,3)
move_and_capture(1,4,4,1)

-- Bishop capture King@(2,3)
move_and_capture(1,4,2,3)

-- Blocked by Q@(4,1)
move_and_capture(2,3,4,1)

