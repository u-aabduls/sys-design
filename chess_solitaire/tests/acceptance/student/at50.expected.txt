  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(Q,1,4)
  # of chess pieces on board: 1
  Game being Setup...
  ...Q
  ....
  ....
  ....
->setup_chess(K,2,3)
  # of chess pieces on board: 2
  Game being Setup...
  ...Q
  ..K.
  ....
  ....
->setup_chess(P,3,2)
  # of chess pieces on board: 3
  Game being Setup...
  ...Q
  ..K.
  .P..
  ....
->setup_chess(Q,4,1)
  # of chess pieces on board: 4
  Game being Setup...
  ...Q
  ..K.
  .P..
  Q...
->start_game
  # of chess pieces on board: 4
  Game In Progress...
  ...Q
  ..K.
  .P..
  Q...
->move_and_capture(1,4,4,1)
  # of chess pieces on board: 4
  Error: Block exists between (1, 4) and (4, 1)
  ...Q
  ..K.
  .P..
  Q...
->move_and_capture(1,4,3,2)
  # of chess pieces on board: 4
  Error: Block exists between (1, 4) and (3, 2)
  ...Q
  ..K.
  .P..
  Q...
->move_and_capture(1,4,4,1)
  # of chess pieces on board: 4
  Error: Block exists between (1, 4) and (4, 1)
  ...Q
  ..K.
  .P..
  Q...
->move_and_capture(1,4,2,3)
  # of chess pieces on board: 3
  Game In Progress...
  ....
  ..Q.
  .P..
  Q...
->move_and_capture(2,3,4,1)
  # of chess pieces on board: 3
  Error: Block exists between (2, 3) and (4, 1)
  ....
  ..Q.
  .P..
  Q...
